src/recaplog

```diff
@@ -20,145 +20,252 @@
-		if [ $EMAIL_ON_ERROR -eq 1 ]; then
-			echo ${PREFIX}" "$2 | mutt -s $EMAIL_ON_ERROR_SUBJECT $EMAIL_ON_ERROR_RECIPIENTS
-		fi
-	fi
+# Function to log messages
+log() {
+  # does not work in a while-loop as spawns a new shell
+  local msg_type=$1
+  shift
+  local log_entry="$*"
+  ## This avoids sending any output to stdout when executed through cron
+  ## is helpful to avoid emails submitted, instead the logs contain the possible ERRORS
+  if ! tty -s; then
+    echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 >> ${LOGFILE}
+  else
+    if [[ ${VERBOSE} ]]; then
+      echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 | tee -a ${LOGFILE}
+    else
+      if [[ ${msg_type} =~ "ERROR" ||
+            ${msg_type} =~ "WARNING" ]]; then
+        echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 | tee -a ${LOGFILE}
+      else
+        echo $( eval ts ) [${msg_type}] ${log_entry} 2>&1 >> ${LOGFILE}
+      fi
+    fi
+  fi
 }
 
-# delete log files older than LOG_EXPIRY days
+# Function to delete log files older than LOG_EXPIRY days
 delete_old_logs() {
-	if [ $LOG_EXPIRY -eq 0 ]; then
-		print_msg INFORMATION "skipping old log file expiration..."
-	else
-		print_msg INFORMATION "deleting log files older than ${LOG_EXPIRY} days..."
-		# delete *.log files
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log" -type f -mtime +$LOG_EXPIRY -exec echo rm {} \;  | tee -a $LOGFILE
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log" -type f -mtime +$LOG_EXPIRY -exec rm {} \;
-		# delete *.log.gz files
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log.gz" -type f -mtime +$LOG_EXPIRY -exec echo rm {} \;  | tee -a $LOGFILE
-		find $BASEDIR -maxdepth 1 -regextype grep -regex "${BASEDIR}/[a-z]*_[0-9]\{8\}-[0-9]\{6\}.log.gz" -type f -mtime +$LOG_EXPIRY -exec rm {} \;
-	fi
+  local prefix_msg='Deleting:'
+  local matched_files=''
+  local empty_dirs=''
+  local count_files=0
+  local count_dirs=0
+  if [[ ${LOG_EXPIRY} -eq 0 ]]; then
+    log INFO "Skipping old log file expiration..."
+  else
+    log INFO "Deleting log files older than ${LOG_EXPIRY} days..."
+    if [[ ${DRYRUN} ]]; then
+      prefix_msg="Would delete:"
+    fi
+    # Deleting files and empty directories if any.
+    # Using now posix-extended regex flavor to ease the matching of logs/gzips/tarballs
+    matched_files=$(
+      find ${BASEDIR} -maxdepth 2 -regextype posix-extended \
+        -type f -mtime +${LOG_EXPIRY} -regex \
+        "^${BASEDIR}/(.*/)?[a-z]+_([a-z]+_)?[0-9]{8}(-[0-9]{6})?\.log(\.tar)?(\.gz)?$"
+    )
+    if [[ -n ${matched_files} ]]; then
+      count_files=$( wc -l <<<"${matched_files}" )
+    fi
+    log INFO "${prefix_msg} ${count_files} log files."
+    if [[ ! ${DRYRUN} ]]; then
+      # Deleting old logs, compressed/tarballs and empty dirs.
+      for matched_file in ${matched_files}; do
+        rm -f ${matched_file}
+      done
```

Comment:
Files with spaces in the path will break this for loop.  You'll need to either backup/reset/restore IFS, or iterate over "while read LINE" instead.  Also, quote the "matched_file" variable so those whitespace characters don't break the rm.